openmicroscopy/openmicroscopy
# remove list of image formats from OMERO code

# What this PR does

This PR removes the obsolete list of image formats from the OMERO object mapping which is easier than needlessly maintaining it. #4837 ensures that all currently valid formats are still available as `Format` instances. The upgrade script ensures that obsolete formats that are used by legacy data remain `Format` instances.

# Testing this PR

test one:

1. Start from a fresh database using `OMERO5.3DEV__12.sql` generated by `omero/bin db script`.
1. Observe that the `format` table is empty.
1. Start the server.
1. Observe that the `format` table is now comprehensively populated.

test two:

1. Start from an older database with some images imported.
1. Run the database upgrade to patch 12.
1. Observe that the `format` table retains entries only for the formats of the imported images.
1. Start the server.
1. Observe that the `format` table is now comprehensively populated.
